root/components/EntityTabs.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * @flow
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+const React = require('react');
+const {l} = require('../static/scripts/common/i18n');
+const Tabs = require('./Tabs.js');
+const EntityTabLink = require('./EntityTabLink');
+const EntityLink = require('../static/scripts/common/components/EntityLink');
+const {ENTITIES} = require('../static/scripts/common/constants');
+
+const tabLinkNames = {
+  artists: l('Artists'),
+  cover_art: l('Cover Art'),
+  discids: l('Disc IDs'),
+  events: l('Events'),
+  fingerprints: l('Fingerprints'),
+  labels: l('Labels'),
+  map: l('Map'),
+  performances: l('Performances'),
+  places: l('Places'),
+  recordings: l('Recordings'),
+  releases: l('Releases'),
+  users: l('Users'),
+  works: l('Works'),
+};
+
+const buildLinkObject = (content, entity, subPath, page, pageName) => {
+  if (!pageName) {
+    pageName = subPath;
+  }
+  return (
+    <EntityTabLink
+      content={content}
+      entity={entity}
+      key={subPath}
+      selected={pageName === page}
+      subPath={subPath}
+    />
+  );
+};
+
+const buildPropertyLinks = (entity: EntityT, page: string) => {
+  const propLinks = [];
+  const entityProperties = ENTITIES[entity.entityType];
+
+  if (entityProperties.custom_tabs) {
+    entityProperties.custom_tabs.map((tab) => {
+      propLinks.push(buildLinkObject(tabLinkNames[tab], entity, tab, page));
+    });
+  }
+
+  if (entityProperties.mbid.relatable === 'dedicated') {
+    propLinks.push(buildLinkObject(l('Relationships'), entity, 'relationships', page));
+  }
+
+  if (entityProperties.aliases) {
+    propLinks.push(buildLinkObject(l('Aliases'), entity, 'aliases', page));
+  }
+
+  if (entityProperties.tags) {
+    propLinks.push(buildLinkObject(l('Tags'), entity, 'tags', page));
+  }
+
+  if (!entityProperties.mbid.no_details) {
+    propLinks.push(buildLinkObject(l('Details'), entity, 'details', page));
+  }
+
+  return propLinks;
+};
+
+const buildLinks = (editTab, entity, page) => {
+  let infoLinks = [buildLinkObject(l('Overview'), entity, 'show', page, 'index')];
+  infoLinks = infoLinks.concat(buildPropertyLinks(entity, page));
+
+  if (typeof editTab === 'string' && editTab === 'default') {
```

Comment:
I went with a slightly different approach. I added a hideEditTab (as opposed to showEditTab, in order to make the hiding more explicit since it was only in special cases that we need to hide the edit tab and I pass in an editTab arg. If editTab is not supplied, it uses the default edit tab and hideEditTab hides the edit tab regardless of the args.)